src/main/src/main/resources/GEOSERVER_DEVELOPER_LOGGING.xml

```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- This log4j configuration file needs to stay here, and is used as a default logging setup   -->
+<!-- during data_dir upgrades and in case the chosen logging config isn't available.            -->
+<Configuration name="DEFAULT_LOGGING" status="fatal" dest="out">
+    <CustomLevels>
+        <CustomLevel name="CONFIG" intLevel="450" />
```

Comment:
That is unclear to me also; I did create the levels in geotools but as I understand it they static methods in geotools may not be called before log4j configuration file is loaded.

It may be okay now that I am creating GeoServerXMLConfiguration by hand; rather than replying on Log4J plugin system (so we could experiment with removing).

Reply:
Okay if I adjust visibility I can remove the custom levels boilerplate from the xml https://osgeo-org.atlassian.net/browse/GEOT-7133